cli/commands/info/index.ts

```diff
@@ -0,0 +1,109 @@
+    agentRegistry: AgentRegistry,
+    agentRegistryContractAddress: string,
+    getFromIpfs: GetFromIpfs,
+    getTransactionReceipt: GetTransactionReceipt,
+): CommandHandler {
+    assertExists(getFromIpfs, 'getFromIpfs')
+
+    return async function info() {
+        const finalAgentId = args.agentId ? args.agentId : agentId;
+
+        assertIsNonEmptyString(finalAgentId, 'agentId');
+
+        const agent = await agentRegistry.getAgent(finalAgentId);
+        const currentState = await agentRegistry.isEnabled(finalAgentId) as boolean
+        const ipfsMetaHash = agent.metadata;
+
+
+        const ipfsData = await getFromIpfs(ipfsMetaHash)
+        printIpfsMetaData(ipfsData,currentState)
+
+        console.log(`Recent Activity (Last ${DAYS_TO_SCAN} days): \n`);
+
+        const blockEventTopicFilters = AGENT_REGESTRY_EVENT_FRAGMENTS
+            .filter(fragment => isRelevantSmartContractEvent(fragment.name))
+            .map(eventFragment => getTopicHashFromEventName(eventFragment.name as StateChangeContractEvent)) as string[];
+
+        const latestBlockNumber = await ethersAgentRegistryProvider.getBlockNumber();
+
+        const network = await ethersAgentRegistryProvider.getNetwork();
+        const { chainId } = network;
+        const { blockTimeInSeconds } = getBlockChainNetworkConfig(chainId);
+
+        const endingBlock = Math.floor(latestBlockNumber - ((DAYS_TO_SCAN * SECONDS_IN_DAY)/(blockTimeInSeconds)));
+
+        const increment = 1000;
+        let startingBlock = latestBlockNumber - (increment * 5);
+
+        const getAgentLogs = async (startBlock: number, endBlock: number) => {
+            return await ethersAgentRegistryProvider.getLogs({
+                address: agentRegistryContractAddress,
+                fromBlock: startBlock,
+                toBlock: endBlock,
+                topics: [[...blockEventTopicFilters], null] // This assumes that each smart contract event is indexed so that the first topic is always an event of interest
+            });
+        }
+        
+        while(startingBlock > endingBlock) {
+            
+            // Get logs in parallel from 5 diffrent block ranges
+            const response = (await Promise.all([
+                getAgentLogs(startingBlock, startingBlock + increment), 
+                getAgentLogs(startingBlock + increment + 1, startingBlock + (increment * 2)),
+                getAgentLogs(startingBlock + (2 * increment) + 1, startingBlock + (increment * 3)),
+                getAgentLogs(startingBlock + (3 * increment) + 1, startingBlock + (increment * 4)),
+                getAgentLogs(startingBlock + (4 * increment) + 1, startingBlock + (increment * 5))
+            ]))
+
+            const logs =  flatten(response)
+
+            for (let log of logs) {
+                const transaction = await getTransactionReceipt(log.transactionHash, true);
+
+                if(transaction.from.toLowerCase() === ipfsData.from.toLowerCase()) {
```

Comment:
Ran into a weird issue when fetching the transaction associated with a log (to make sure the address sending the transaction is the same as the bot owner) the address is returned in all lower case. Ipfs returns the address correctly though.

Reply:
An edge case here is filtering logs is by ownerId (can't find any agentId's in the logs or transactions) so if the same owner creates 2 bots, not sure if there is a way to filter logs further by agentId.